fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexValidator.java

```diff
@@ -44,5 +46,49 @@ public void validate(@Nonnull MetaDataValidator metaDataValidator) {
         for (RecordType recordType : metaDataValidator.getRecordMetaData().recordTypesForIndex(index)) {
             LuceneIndexExpressions.validate(index.getRootExpression(), recordType.getDescriptor());
         }
+
+        validateIndexOptions(index);
+    }
+
+    @VisibleForTesting
+    public static void validateIndexOptions(@Nonnull Index index) {
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.LUCENE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.AUTO_COMPLETE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAutoCompleteExcludedFields(index);
+    }
+
+    private static void validateAnalyzerNamePerFieldOption(@Nonnull String optionKey, @Nonnull Index index) {
+        String analyzerNamePerFieldOption = index.getOption(optionKey);
+        if (analyzerNamePerFieldOption != null) {
+            analyzerNamePerFieldOption = analyzerNamePerFieldOption.strip();
+            if (analyzerNamePerFieldOption.startsWith(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE)
+                    || analyzerNamePerFieldOption.startsWith(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS)
+                    || analyzerNamePerFieldOption.endsWith(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE)
+                    || analyzerNamePerFieldOption.endsWith(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS)) {
+                throw new MetaDataException("Index " + index.getName() + " has invalid " + optionKey + ": " + analyzerNamePerFieldOption);
+            }
+            String[] keyValuePairs = analyzerNamePerFieldOption.split(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS);
+            for (String keyValue : keyValuePairs) {
+                if (keyValue.indexOf(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE) < 1
+                        || keyValue.indexOf(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE) > keyValue.length() - 2
+                        || keyValue.indexOf(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE) != keyValue.lastIndexOf(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE)) {
+                    throw new MetaDataException("Index " + index.getName() + " has invalid " + optionKey + ": " + analyzerNamePerFieldOption);
```

Comment:
```suggestion
                    throw new MetaDataException("Index " + index.getName() + " has invalid option value " + optionKey + ": " + analyzerNamePerFieldOption);
```

Likewise for the other `MetaDataExceptions` that are thrown in this class

Reply:
Fixed it